src/main/java/org/opentripplanner/service/worldenvelope/model/WorldEnvelope.java

```diff
@@ -0,0 +1,124 @@
+package org.opentripplanner.service.worldenvelope.model;
+
+import java.io.Serializable;
+import java.util.Optional;
+import org.opentripplanner.framework.geometry.WgsCoordinate;
+import org.opentripplanner.framework.tostring.ToStringBuilder;
+
+/**
+ * This class calculates borders of envelopes that can be also on 180th meridian The same way as it
+ * was previously calculated in GraphMetadata constructor
+ */
+public class WorldEnvelope implements Serializable {
+
+  private static final WorldEnvelope DEFAULT = new WorldEnvelope(
+    WgsCoordinate.GREENWICH.add(-1d, -1d),
+    WgsCoordinate.GREENWICH.add(1d, 1d),
+    WgsCoordinate.GREENWICH
+  );
+
+  private final WgsCoordinate lowerLeft;
+  private final WgsCoordinate upperRight;
+  private final WgsCoordinate meanCenter;
+  private final WgsCoordinate transitMedianCenter;
+
+  private WorldEnvelope(
+    WgsCoordinate lowerLeft,
+    WgsCoordinate upperRight,
+    WgsCoordinate transitMedianCenter
+  ) {
+    this.transitMedianCenter = transitMedianCenter;
+    this.lowerLeft = lowerLeft;
+    this.upperRight = upperRight;
+    this.meanCenter = calculateMeanCenter(lowerLeft, upperRight);
+  }
+
+  WorldEnvelope(
+    double lowerLeftLatitude,
+    double lowerLeftLongitude,
+    double upperRightLatitude,
+    double upperRightLongitude,
+    WgsCoordinate transitMedianCenter
+  ) {
+    this(
+      new WgsCoordinate(lowerLeftLatitude, lowerLeftLongitude),
+      new WgsCoordinate(upperRightLatitude, upperRightLongitude),
+      transitMedianCenter
+    );
+  }
+
+  public static WorldEnvelopeBuilder of() {
+    return new WorldEnvelopeBuilder();
+  }
+
+  /**
+   * Return a small square envelope around Greenwich to use as a default value. This allows
+   * us to always provide an envelope, even when there is no data.
+   */
+  public static WorldEnvelope defaultEnvelope() {
+    return DEFAULT;
+  }
```

Comment:
Is there a reason why to do this, rather than returning some kind of "empty envelope", or even an optional value? I would assume this might cause headaches when trying to debug this functionality, when the client returns a clearly erroneous value.